### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by krasnowsky**

### CODE REVIEW

File: WebBackend/app.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            features[field] = data.get(field)
+        location_features = _get_location_data(
+            features['latitude'],
+            features['longitude']
+        )
+        if location_features['CountryId'] != 'PL':
+            return {'error': 'Please drop pin in boarders of Poland'}, 400
+        features['population'] = location_features['Population']
+        ready_data = encoder.transform(features)
+        prediction = model.predict(ready_data)
```
Comment (danielbaraniak) [inline WebBackend/app.py]: In new ml implementation #2 , model contains whole pipeline with encoding, so it's not nessesseary to do `encoder.transform(features)`. However, `model.predict(ready_data)` would need `pandas.DataFrame`.
```suggestion
        ready_data = pd.DataFrame({k: [v] for k, v in features.items()})
        prediction = model.predict(ready_data)
```

File: WebBackend/app.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+limiter = Limiter(
+    app,
+    key_func=get_remote_address,
+    default_limits=['1/5seconds']
+)
+
+def _get_location_data(latitude, longitude):
+    url = 'https://geocodeapi.p.rapidapi.com/GetNearestCities'
+    location_fields = ['Population', 'CountryId']
```
Comment (danielbaraniak) [inline WebBackend/app.py]: I think `City` (`city_name`) also could be included since we already getting it in response.

**Review by danielbaraniak — APPROVED** [review]

File: WebBackend/app.py (hunk 1 above)
Comment (krasnowsky) [reply WebBackend/app.py]: Corrected

File: WebBackend/app.py (hunk 2 above)
Comment (krasnowsky) [reply WebBackend/app.py]: Added field city_name

**Comment by krasnowsky** [comment]
@230443 Added requested changes

**Review by danielbaraniak — COMMENTED** [review]
Should work with a model from #2. The problem is with #4. Is it possible to make the `ML` module available for import, for example by starting app.py in script: https://github.com/230443/Sprinters/blob/ml-src/launch.py? Alternatively we can add this at the beginning of app.py:

```python3
import sys

if ".." not in sys.path:
    sys.path.insert(0, "..")
```

**Pull request merged**